### DISCUSSION TITLE
Standard Deviation Channels

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#368] [Indicators]
Community discussion for [Standard Deviation Channels](https://dotnet.stockindicators.dev/indicators/StdDevChannels/#content)

**Comment by segeyros** [comment]
Hi, Dave!
The indicator GetStdDevChannels behaves strangely.

I use this commands:
     var connorRsiResult = quotes.GetConnorsRsi(3, 2, 100).ToArray();
     var crsiChannelsResult = connorRsiResult.GetStdDevChannels(10, 1.0).ToArray();

Then I form the arrays:
                t[i] = crsiChannelsResult[i].Date;
                channel[i] = crsiChannelsResult[i].UpperChannel;

Then I output the array to the graph (see graph 1 in the attached file).
Then I add another candle to the "quote", passing it through "GetStdDevChannels", and get a completely different graph (see graph 2 in the attached file). Perhaps this is due to the fact that GetStdDevChannels processes the Quote from the end of the array, and not at first as expected.
What should I do?

![indicator](https://github.com/DaveSkender/Stock.Indicators/assets/83514404/fddd6369-480a-4874-b5ad-44da9949a6de)

**Reply by DaveSkender** [reply]
If you add a single quote to `quotes`, you’ll have to replay all the steps to rebuild the whole dataset.  It’s not going to auto-calculate any incremental values.   We’ll add [incremental and streaming features](https://github.com/DaveSkender/Stock.Indicators/discussions/1018) in v3

**Reply by DaveSkender** [reply]
You likely have done that.  Peeking at [the docs](https://dotnet.stockindicators.dev/indicators/StdDevChannels/#response) to refresh my memory, I also see a note that these channels repaint.

<img alt="image" src="https://github.com/DaveSkender/Stock.Indicators/assets/8432125/b537aed7-1012-4141-b834-4c66be2ece30">

**Reply by DaveSkender** [reply]
Maybe try [Bollinger Bands](https://dotnet.stockindicators.dev/indicators/BollingerBands/#content) instead of StdDev Channels, it uses a moving average model that won’t suffer from window positions.

**Reply by segeyros** [reply]
Thanks, I understand you. Hopefully in version 3 I will be able to handle this in a stream without fluctuation. Actually, my task is to get the direction of change of the CRSI indicator. And, ultimately, the velocity of this direction change. This velocity is expected to be a good help when making program decisions about buying and selling on the stock exchange.

**Reply by DaveSkender** [reply]
Given the volatility of CRSI, consider [slope](https://dotnet.stockindicators.dev/indicators/Slope/#content) for direction and velocity, idea:

```csharp
var results = quotes
  .GetConnorsRsi(N)
  .GetSlope(N/2);
```

**Reply by segeyros** [reply]
I'll try

**Reply by segeyros** [reply]
<Maybe try [Bollinger Bands] instead of StdDev Channels>
I tried it.
it does not give the desired result, unlike StdDev Channel, which gives a perfect result, which is shown on the graph:
![image](https://github.com/DaveSkender/Stock.Indicators/assets/83514404/6272f791-fb4a-4d87-a4b1-514bcebbf592)
And now, after trying out "Slope" (similar to StdDev Channels but without history), I understand why StdDev Channel behaves so changeable
